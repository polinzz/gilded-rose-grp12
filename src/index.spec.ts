import runGoldenMaster from "jest-golden-master";
import { GildedRose, Item } from "./index";

// Définir la fonction generateCombinations
function generateCombinations(
    names: string[],
    sellIN: number[],
    quantity: number[]
) {
    const combinations: { name: string; sellIn: number; quality: number }[] = [];

    for (const name of names) {
        for (const si of sellIN) {
            for (const q of quantity) {
                combinations.push({ name, sellIn: si, quality: q });
            }
        }
    }

    return combinations;
}

test("My first scenario", async () => {
    await runGoldenMaster(async () => {
        const items = [new Item("TOTO", 10, 10)];
        const gilderRose = new GildedRose(items);
        console.log(gilderRose.updateQuality());
    });
});

test("My second scenario", async () => {
    await runGoldenMaster(async () => {
        const names = [
            '+5 Dexterity Vest',
            'Aged Brie',
            'Elixir of the Mongoose',
            'Sulfuras, Hand of Ragnaros',
            'Backstage passes to a TAFKAL80ETC concert',
            'Conjured Mana Cake',
        ];
        const sellIN = [10, 2, 5, 0, 15, 3, -1];
        const quantity = [45];

        // Générer toutes les combinaisons possibles
        const allCombinations = generateCombinations(names, sellIN, quantity);

        // Exécuter le test pour chaque combinaison
        for (const combination of allCombinations) {
            const items = [new Item(combination.name, combination.sellIn, combination.quality)];

            const gilderRose = new GildedRose(items);
            console.log(gilderRose.updateQuality());

        }
    });
});
