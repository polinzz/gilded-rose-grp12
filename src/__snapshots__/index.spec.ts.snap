// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My first scenario 1`] = `
[
  [
    Item {
      "name": "TOTO",
      "quality": 9,
      "sellIn": 9,
    },
  ],
]
`;

exports[`My second scenario 1`] = `
[
  [
    Item {
      "name": "+5 Dexterity Vest",
      "quality": 44,
      "sellIn": 9,
    },
  ],
  [
    Item {
      "name": "+5 Dexterity Vest",
      "quality": 44,
      "sellIn": 1,
    },
  ],
  [
    Item {
      "name": "+5 Dexterity Vest",
      "quality": 44,
      "sellIn": 4,
    },
  ],
  [
    Item {
      "name": "+5 Dexterity Vest",
      "quality": 43,
      "sellIn": -1,
    },
  ],
  [
    Item {
      "name": "+5 Dexterity Vest",
      "quality": 44,
      "sellIn": 14,
    },
  ],
  [
    Item {
      "name": "+5 Dexterity Vest",
      "quality": 44,
      "sellIn": 2,
    },
  ],
  [
    Item {
      "name": "+5 Dexterity Vest",
      "quality": 43,
      "sellIn": -2,
    },
  ],
  [
    Item {
      "name": "Aged Brie",
      "quality": 46,
      "sellIn": 9,
    },
  ],
  [
    Item {
      "name": "Aged Brie",
      "quality": 46,
      "sellIn": 1,
    },
  ],
  [
    Item {
      "name": "Aged Brie",
      "quality": 46,
      "sellIn": 4,
    },
  ],
  [
    Item {
      "name": "Aged Brie",
      "quality": 47,
      "sellIn": -1,
    },
  ],
  [
    Item {
      "name": "Aged Brie",
      "quality": 46,
      "sellIn": 14,
    },
  ],
  [
    Item {
      "name": "Aged Brie",
      "quality": 46,
      "sellIn": 2,
    },
  ],
  [
    Item {
      "name": "Aged Brie",
      "quality": 47,
      "sellIn": -2,
    },
  ],
  [
    Item {
      "name": "Elixir of the Mongoose",
      "quality": 44,
      "sellIn": 9,
    },
  ],
  [
    Item {
      "name": "Elixir of the Mongoose",
      "quality": 44,
      "sellIn": 1,
    },
  ],
  [
    Item {
      "name": "Elixir of the Mongoose",
      "quality": 44,
      "sellIn": 4,
    },
  ],
  [
    Item {
      "name": "Elixir of the Mongoose",
      "quality": 43,
      "sellIn": -1,
    },
  ],
  [
    Item {
      "name": "Elixir of the Mongoose",
      "quality": 44,
      "sellIn": 14,
    },
  ],
  [
    Item {
      "name": "Elixir of the Mongoose",
      "quality": 44,
      "sellIn": 2,
    },
  ],
  [
    Item {
      "name": "Elixir of the Mongoose",
      "quality": 43,
      "sellIn": -2,
    },
  ],
  [
    Item {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 45,
      "sellIn": 10,
    },
  ],
  [
    Item {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 45,
      "sellIn": 2,
    },
  ],
  [
    Item {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 45,
      "sellIn": 5,
    },
  ],
  [
    Item {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 45,
      "sellIn": 0,
    },
  ],
  [
    Item {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 45,
      "sellIn": 15,
    },
  ],
  [
    Item {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 45,
      "sellIn": 3,
    },
  ],
  [
    Item {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 45,
      "sellIn": -1,
    },
  ],
  [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 47,
      "sellIn": 9,
    },
  ],
  [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 48,
      "sellIn": 1,
    },
  ],
  [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 48,
      "sellIn": 4,
    },
  ],
  [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -1,
    },
  ],
  [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 46,
      "sellIn": 14,
    },
  ],
  [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 48,
      "sellIn": 2,
    },
  ],
  [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -2,
    },
  ],
  [
    Item {
      "name": "Conjured Mana Cake",
      "quality": 44,
      "sellIn": 9,
    },
  ],
  [
    Item {
      "name": "Conjured Mana Cake",
      "quality": 44,
      "sellIn": 1,
    },
  ],
  [
    Item {
      "name": "Conjured Mana Cake",
      "quality": 44,
      "sellIn": 4,
    },
  ],
  [
    Item {
      "name": "Conjured Mana Cake",
      "quality": 43,
      "sellIn": -1,
    },
  ],
  [
    Item {
      "name": "Conjured Mana Cake",
      "quality": 44,
      "sellIn": 14,
    },
  ],
  [
    Item {
      "name": "Conjured Mana Cake",
      "quality": 44,
      "sellIn": 2,
    },
  ],
  [
    Item {
      "name": "Conjured Mana Cake",
      "quality": 43,
      "sellIn": -2,
    },
  ],
]
`;
